.github/workflows/bump-version.yaml

```
@@ -0,0 +1,186 @@
+# This workflow does version management, bumping the patch version on the main branch after a PR is merged.
+
+# Requirements:
+# - It needs a GitHub secret named WORKFLOW_PAT with a personal access token that has workflow permissions. The user that owns the
+#   PAT must be allowed to push to the main branch of the repo. That usually means the user has to be a repo admin, and admins must
+#   be excepted from branch protection rules on main.
+# - Another workflow must consume the prereleases created by this one. That workflow should build and publish the release.
+
+# Terminology:
+# - In semver, everything before the optional "-" is the version, and everything after it is the prerelease version.
+# - In this workflow, we can operate on the release part of the semver, or on the prerelease part of the semver.
+# - In GitHub, we create a "release" object that can optionally have the "prerelease" flag set to true. The "docker.yaml" workflow
+#   triggers on release events.
+# I'm sorry the terminology is bad.
+
+# Customize:
+# It can work in either release mode or prerelease mode. That controls whether it increments the release version or the prerelease
+# version. Here's an example of how the two modes transform the current version into the release version and bumped version:
+# | mode    | release     | prerelease  |
+# | current | 1.2.3-pre   | 1.2.3-pre.4 |
+# | release | 1.2.3       | 1.2.3-rc.4  |
+# | bumped  | 1.2.4-pre   | 1.2.3-pre.5 |
+# The default mode is "release". To get "prerelease" mode, modify /env/MODE.
+
+# Details:
+# With our typical triggers, this workflow is triggered by merge to main, or by manual workflow_dispatch.
+# 1. Make a release version. Depending on whether we're in "release" or "prerelease" mode:
+#    - release: Remove the -prerelease info from the semver. (1.2.3-pre -> 1.2.3)
+#    - prerelease: Remove any non-numeric characters from the prerelease part of the semver. (1.2.3-pre.4 -> 1.2.3-4)
+# 2. Commit, push, tag, and create a GitHub prerelease.
+# 3. Bump the version, and add `-pre` or `.pre` as a pre-release identifier.
+#    - release: 1.2.3 -> 1.2.4-pre
+#    - prerelease: 1.2.3-4 -> 1.2.3-pre.5
+# 4. Commit and push.
+# It parses the version info from files in the repo like Cargo.toml, package.json, or version.sbt. It updates those files when it
+# bumps the version.
+# Sanity checks ensure it's a valid semver, and it's not 0.0.0.
+
+name: Bump Version
+
+on:
+  workflow_call:
+    inputs:
+      version:
+        description: New semver release version.
+        type: string
+        required: true
```

```suggestion
        description: Override new semver release version.
        type: string
        required: false
```

If it's triggered by `workflow_dispatch`, we might have a `version` input. But if it's triggered by push to `main`, we won't. In that case, the workflow will determine a new version by using `bump-version.bump.sh`.

---

I reworded the description a bit, but I think it should still be required to make sure that the caller passes their `inputs.version` through correctly (or they explicitly pass empty string if it doesn't exist)